lib/system/memory.nim

```diff
@@ -51,4 +51,33 @@ proc nimCStrLen*(a: cstring): int {.compilerproc, nonReloadable, inline.} =
       a = cast[ptr byte](cast[uint](a) + 1)
       inc result
 
+proc nimMoveMem*(dest, source: pointer, size: Natural) {.nonReloadable, compilerproc, inline.} =
+  when useLibC:
+    c_memmove(dest, source, cast[csize_t](size))
```

Comment:
I will never understand what the thought process was behind `dest` being the first argument for `memmove`.